pkgs/applications/editors/emacs/elisp-packages/manual-packages/hsc3/default.nix

```
@@ -1,21 +1,23 @@
-{ lib, trivialBuild, fetchurl, haskell-mode }:
+{
+  melpaBuild,
+  haskell-mode,
+  haskellPackages,
+}:
 
-trivialBuild rec {
+let
+  inherit (haskellPackages) hsc3;
+in
+melpaBuild {
   pname = "hsc3-mode";
-  version = "0.15";
+  ename = "hsc3";
+  inherit (hsc3) src version;
 
-  src = fetchurl {
-    url = "mirror://hackage/hsc3-${version}/hsc3-${version}.tar.gz";
-    sha256 = "2f3b15655419cf8ebe25ab1c6ec22993b2589b4ffca7c3a75ce478ca78a0bde6";
-  };
+  files = ''("emacs/*.el")'';
 
   packageRequires = [ haskell-mode ];
 
-  sourceRoot = "hsc3-${version}/emacs";
-
   meta = {
-    homepage = "http://rd.slavepianos.org/?t=hsc3";
+    inherit (hsc3.meta) homepage license;
     description = "hsc3 mode package for Emacs";
```

AFAICR, guidelines require upper case for the first char.

```suggestion
    description = "Emacs mode for hsc3";
```